pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -2170,6 +2170,14 @@ protected void internalGetMessageById(AsyncResponse asyncResponse, long ledgerId
             validateReadOperationOnTopic(authoritative);
             PersistentTopic topic = (PersistentTopic) getTopicReference(topicName);
             ManagedLedgerImpl ledger = (ManagedLedgerImpl) topic.getManagedLedger();
+            if (null == ledger.getLedgerInfo(ledgerId).get()) {
```

Comment:
Is it fine to perform a blocking method call here?

Reply:
Agree, this should be done asynchronously

Reply:
It seems that theres no time-consuming operation, any other point needs to consider?

Reply:
It's not about time. A Netty threads should never be blocked.

Reply:
It seems it is not run in a netty thread. btw, https://github.com/apache/pulsar/pull/11912 is better to fix this bug.